### PR TITLE
Move project to Boxed.Framework.

### PR DESCRIPTION
**Pull request by VictorioBerra**
Remove template specific code.
Remove unused dependencies.
Update .csproj with PropertyGroups needed for creating a package.

### CODE REVIEW

**Comment by VictorioBerra** [comment]
ref https://github.com/Dotnet-Boxed/Templates/issues/357

File: Source/Boxed.DotnetNewTest/Constants/GraphQlQuery.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
+namespace Boxed.Templates.FunctionalTest.Constants
+{
+    public static class GraphQlQuery
```
Comment (RehanSaeed) [inline Source/Boxed.DotnetNewTest/Constants/GraphQlQuery.cs]: All of this GraphQL stuff should not be added to this package.

File: Source/Boxed.DotnetNewTest/Framework/AssemblyResolver.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using System.Runtime.Loader;
+    using Microsoft.Extensions.DependencyModel;
+    using Microsoft.Extensions.DependencyModel.Resolution;
+
+    internal sealed class AssemblyResolver : IDisposable
```
Comment (RehanSaeed) [inline Source/Boxed.DotnetNewTest/Framework/AssemblyResolver.cs]: I run functional tests by calling the dotnet CLI commands.

However, there is another way I have been experimenting with but have failed to get working so far. This works as follows:

1. Build the template project.
2. Load the built template assembly using the new `AssemblyLoadContext` API in .NET Core.
3. Uses the `TestHost` API in ASP.NET Core to run the API/Swagger/GraphQL/Orleans in memory.
4. Run tests code against it e.g. call the API running in memory.

In theory this will be a lot faster. The downside is that I can't yet get the assembly loading stuff to work reliably. Also .NET Core 3 has support for unloading assemblies at the end of the test which is also important.

To sum up, this code does not work but I want it to.

File: Source/Boxed.DotnetNewTest/Framework/TempDirectoryExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+namespace Boxed.Templates.FunctionalTest
```
Comment (RehanSaeed) [inline Source/Boxed.DotnetNewTest/Framework/TempDirectoryExtensions.cs]: All namespaces should be changed to `Boxed.DotnetNewTest/*`

File: Source/Boxed.DotnetNewTest/Framework/AssemblyResolver.cs (hunk 2 above)
Comment (VictorioBerra) [reply Source/Boxed.DotnetNewTest/Framework/AssemblyResolver.cs]: I like this idea. Maybe its worth tabling this as a GitHub issue until core 3 is released in September?

Comment (RehanSaeed) [reply Source/Boxed.DotnetNewTest/Framework/AssemblyResolver.cs]: Yes, thats a good idea.

**Comment by RehanSaeed** [comment]
FYI, there is a problem with running `DotnetRun` on CI servers. You need a developer certificate installed using the `dotnet dev-certs` command line tool. However, this tool shows a UI and I can't think of a way to work around this problem. See [this StackOverflow question](https://stackoverflow.com/questions/57221762/how-to-install-dotnet-dev-certs-certificate-on-a-ci-server?noredirect=1#comment100947876_57221762) for more info.

I had all tests skipped for some time because of this. Tried fixing it last week but failed. `DotnetRestore` and `DotnetBuild` all work fine. If I can't get it to work, I might skip all tests that run the apps and only run `DotnetRestore` and `DotnetBuild` tests on CI servers. Kind of sad.

File: Source/Boxed.DotnetNewTest/Boxed.DotnetNewTest.csproj [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  <PropertyGroup Label="Build">
+    <TargetFrameworks>netcoreapp2.2</TargetFrameworks>
+    <LangVersion>latest</LangVersion>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <CodeAnalysisRuleSet>../../MinimumRecommendedRulesWithStyleCop.ruleset</CodeAnalysisRuleSet>
+  </PropertyGroup>
+
+  <PropertyGroup Label="Package">
+    <VersionPrefix>3.2.0</VersionPrefix>
```
Comment (RehanSaeed) [inline Source/Boxed.DotnetNewTest/Boxed.DotnetNewTest.csproj]: This should be `1.0.0`.

**Comment by RehanSaeed** [comment]
FYI, tests are running in the Templates repo. I turned off all tests that use `dotnet run` on the CI server for now unfortunately. I'm going to create an issue in the tooling repo to see if the dotnet team has any answers.

Also, if you don't have time to write a quick addition to the ReadMe, that's fine. I'll get to it eventually.

**Comment by RehanSaeed** [comment]
https://github.com/aspnet/AspNetCore/issues/12680

**Comment by VictorioBerra** [comment]
Added some info to the README.

**Comment by VictorioBerra** [comment]
> [aspnet/AspNetCore#12680](https://github.com/aspnet/AspNetCore/issues/12680)

Can we just  run the dev-certs command on the CI machine itself interactively and confirm the trust dialog and move forward with this?

**Comment by RehanSaeed** [comment]
We have an answer. We can export the certificate and then use C# to install it. I'm wondering if this should be a method call as part of the test framework or if it should be a build step (probably the former?).

```bash
dotnet dev-certs https --export-path <path>
```

**Comment by VictorioBerra** [comment]
> We have an answer. We can export the certificate and then use C# to install it. I'm wondering if this should be a method call as part of the test framework or if it should be a build step (probably the former?).
> 
> ```shell
> dotnet dev-certs https --export-path <path>
> ```

You still need to use powershell to manually install it and avoid the Windows prompt if I understand correctly, right? Are you going to automate this?

I hate to bake CI/CD specific stuff into the test framework but this seems to be an edge case. Could we leverage cake to do this for us somehow? Or are we going to have to use the WebApplicationFactory to tap into the Startup pipeline?

**Comment by RehanSaeed** [comment]
I fixed an issue with the code in this PR https://github.com/Dotnet-Boxed/Templates/commit/296d45676cd4c7c47462c7f13f54cf60ee069071. See the comments. If we update this PR, I'm happy to get this merged.

**Comment by VictorioBerra** [comment]
@RehanSaeed I have cherry picked your changed from https://github.com/Dotnet-Boxed/Templates/pull/364 does this finalize the PR?

Also you should note there are various unused parameters and lines in ProcessAssert I saw when copying over code. For example `GetAndWriteMessage` never uses `filename`, `arguments`, or  `workingDirectory`

**Comment by RehanSaeed** [comment]
Thanks for you work @VictorioBerra! Added to the contributions page.

**Comment by RehanSaeed** [comment]
Now published manually. Still trying to figure out the build.
